### PR TITLE
only updat Source lines when source_server is not-empty

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: obal/data/roles/update_spec_file/tasks/main.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     replace: '\g<prefix>{{ source_server }}\g<filename>'
   when:
     - source_server is defined
+    - source_server != ""
```
Comment (ehelms) [inline obal/data/roles/update_spec_file/tasks/main.yml:9]: More a question than anything, when does one use `is not` vs `!=` ?

Comment (evgeni) [reply obal/data/roles/update_spec_file/tasks/main.yml:9]: I wouldn't know :)

Comment (evgeni) [reply obal/data/roles/update_spec_file/tasks/main.yml:9]: In Python, `is` and `==` are different (`is` compares identity, `==` compares values). But I am not sure it matters in Ansible.

Comment (evgeni) [reply obal/data/roles/update_spec_file/tasks/main.yml]: and ansible-lint told me to write it otherwise anyways :woman_shrugging:

**Pull request merged**